frontend/src/components/VaultCreationProgress.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <div class="flex items-center">
+    <template v-for="(step, index) in steps" :key="step">
+      <div class="flex items-center">
+        <div
+          class="relative z-10 flex h-7 w-7 items-center justify-center rounded-full border-3 cursor-default"
+          :class="{
+            'bg-primary border-primary text-white': index < currentStep,
+            'bg-white border-gray-300 text-black': index > currentStep,
+            'bg-white border-primary text-white': index === currentStep
+          }"
+          :title="tooltipForStep(step)"
+        >
+          <template v-if="index < currentStep">
+            <CheckIcon class="text-white" />
+          </template>
+          <template v-else>
+            <span class="h-1.5 w-1.5 rounded-full block" :class="index === currentStep ? 'bg-primary' : 'bg-gray-300'"/>
+          </template>
+        </div>
+        <div
+          v-if="index < steps.length - 1"
+          class="w-10 sm:w-12 h-1"
```

Comment:
I honestly don't know if this is correct here, but my goal was to have the same thickness between the border from before and the "line", basically 2px.

```suggestion
          class="w-10 sm:w-12 h-0.5"
```